irc_send.c

```diff
@@ -213,13 +213,7 @@ void irc_send_names(irc_channel_t *ic)
 			*namelist = 0;
 		}
 
-		if (icu->flags & IRC_CHANNEL_USER_OP) {
-			strcat(namelist, "@");
-		} else if (icu->flags & IRC_CHANNEL_USER_HALFOP) {
-			strcat(namelist, "%");
-		} else if (icu->flags & IRC_CHANNEL_USER_VOICE) {
-			strcat(namelist, "+");
-		}
+		namelist[strlen(namelist)] = irc_channel_user_get_prefix(icu);
```

Comment:
This does terrify me somewhat though the string should I think be fully \0-initialised by assigning "" to it, does glib maybe have an append_char function?